src/neo-vm/Types/Integer.cs

```diff
@@ -39,17 +39,24 @@ public override bool GetBoolean()
             return !value.IsZero;
         }
 
-        public override byte[] GetByteArray()
+        public override ReadOnlyMemory<byte> GetByteArray()
         {
             return value.ToByteArray();
         }
 
+#if NETCOREAPP
+        public override int GetByteLength()
+        {
+            return value.GetByteCount();
+        }
+#else
         private int _length = -1;
         public override int GetByteLength()
         {
             if (_length == -1)
-                _length = value.ToByteArray().Length;
+                _length = value.GetByteCount();
```

Comment:
whi with `NETCOREAPP` is not cached?

Reply:
In netcoreapp2.1+, `GetByteCount()` is a member method of `BigInteger`, and it returns the length directly from the internal field.